File: CPP/PetShop.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

// Kelas PetShop untuk menyimpan informasi produk di toko hewan peliharaan
class PetShop {
protected: 
    string id;            // ID produk
    string nama_produk;   // Nama produk
    int harga_produk;     // Harga produk
    int stok_produk;      // Stok produk

public:
    // Constructor default
    PetShop() : id(""), nama_produk(""), harga_produk(0), stok_produk(0) {}

    // Constructor dengan parameter untuk menginisialisasi atribut
    PetShop(string id, string nama_produk, int harga_produk, int stok_produk) {
        this->id = id;               // Mengatur ID produk
        this->nama_produk = nama_produk; // Mengatur nama produk
        this->harga_produk = harga_produk; // Mengatur harga produk
        this->stok_produk = stok_produk;   // Mengatur stok produk
    }

    // Setter untuk ID produk
    void setId(string id) { 
        this->id = id; 
    }
    
    // Getter untuk mendapatkan ID produk
    string getId() { 
        return id; 
    }

    // Setter untuk nama produk
    void setNamaProduk(string nama_produk) { 
        this->nama_produk = nama_produk; 
    }

    // Getter untuk mendapatkan nama produk
    string getNamaProduk() { 
        return nama_produk; 
    }

    // Setter untuk harga produk
    void setHargaProduk(int harga_produk) { 
        this->harga_produk = harga_produk; 
    }

    // Getter untuk mendapatkan harga produk
    int getHargaProduk() { 
        return harga_produk; 
    }

    // Setter untuk stok produk
    void setStokProduk(int stok_produk) { 
        this->stok_produk = stok_produk; 
    }

    // Getter untuk mendapatkan stok produk
    int getStokProduk() { 
        return stok_produk; 
    }

    // Fungsi untuk menampilkan informasi produk
    void display(int id_max, int nama_max, int harga_max, int stok_max) {
        cout << id << string(id_max - id.size(), ' ') // Menampilkan ID dengan padding
             << nama_produk << string(nama_max - nama_produk.size(), ' ') // Menampilkan nama produk dengan padding
             << harga_produk << string(harga_max - to_string(harga_produk).size(), ' ') // Menampilkan harga produk dengan padding
             << stok_produk << string(stok_max - to_string(stok_produk).size(), ' '); // Menampilkan stok produk dengan padding
    }
};

File: JAVA/PetShop.java
public class PetShop {
    // Deklarasi atribut untuk menyimpan informasi produk
    protected String id;           
    protected String namaProduk;   
    protected int hargaProduk;     
    protected int stokProduk;      

    // Constructor default untuk inisialisasi atribut dengan nilai awal
    public PetShop() {
        this.id = "";              // Inisialisasi ID dengan string kosong
        this.namaProduk = "";      // Inisialisasi nama produk dengan string kosong
        this.hargaProduk = 0;      // Inisialisasi harga dengan nilai 0
        this.stokProduk = 0;       // Inisialisasi stok dengan nilai 0
    }

    // Constructor dengan parameter untuk mengisi nilai atribut saat objek dibuat
    public PetShop(String id, String namaProduk, int hargaProduk, int stokProduk) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.hargaProduk = hargaProduk;
        this.stokProduk = stokProduk;
    }

    // Getter untuk mendapatkan nilai atribut id
    public String getId() {
        return id;
    }

    // Setter untuk mengubah nilai atribut id
    public void setId(String id) {
        this.id = id;
    }

    // Getter untuk mendapatkan nilai atribut namaProduk
    public String getNamaProduk() {
        return namaProduk;
    }

    // Setter untuk mengubah nilai atribut namaProduk
    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    // Getter untuk mendapatkan nilai atribut hargaProduk
    public int getHargaProduk() {
        return hargaProduk;
    }

    // Setter untuk mengubah nilai atribut hargaProduk
    // Bisa ditambahkan validasi agar harga tidak negatif
    public void setHargaProduk(int hargaProduk) {
        if (hargaProduk >= 0) { // Validasi agar harga tidak negatif
            this.hargaProduk = hargaProduk;
        } else {
            System.out.println("Harga produk tidak boleh negatif!");
        }
    }

    // Getter untuk mendapatkan nilai atribut stokProduk
    public int getStokProduk() {
        return stokProduk;
    }

    // Setter untuk mengubah nilai atribut stokProduk
    // Bisa ditambahkan validasi agar stok tidak negatif
    public void setStokProduk(int stokProduk) {
        if (stokProduk >= 0) { // Validasi agar stok tidak negatif
            this.stokProduk = stokProduk;
        } else {
            System.out.println("Stok produk tidak boleh negatif!");
        }
    }

    // Method untuk menampilkan informasi produk dalam format tabel
    public void display(int idMax, int namaMax, int hargaMax, int stokMax) {
        // Menampilkan data produk dengan format lebar kolom yang ditentukan
        System.out.printf("%-" + idMax + "s %-" + namaMax + "s %-" + hargaMax + "d %-" + stokMax + "d\n", 
                          id, namaProduk, hargaProduk, stokProduk);
    }
}
